### PR TITLE
feat: integrated the heat map

### PR DESCRIPTION
**Pull request by gis727**
Note: requires #97.

Added the heat map template and integrated the heat map in the repository component.

[Deployed](https://flaky-dev-staging.web.app/org/nodejs/node) at the time of writing.

### CODE REVIEW

File: packages/frontend/src/app/heat-map/heat-map.component.css [hunk 1]
```diff
@@ -13,3 +13,71 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
...
+
+.flaky {
+    background-color: #fff4e0;
+}
+.failing {
+    background-color: #ffdede;
+}
+.success {
+    background-color: greenyellow;
+}
```
Comment (gis727) [inline packages/frontend/src/app/heat-map/heat-map.component.css:83]: [flex-layout](https://github.com/angular/flex-layout) will be integrated in a different PR to get rid of some of this css.

File: packages/frontend/src/app/heat-map/heat-map.component.html [hunk 2]
```diff
@@ -13,3 +13,35 @@
...
+      <mat-nav-list>
+      <cdk-virtual-scroll-viewport itemSize="50" id="builds-viewport">
+          <p id="build-view-date">Builds on {{selectedBatchDate}}</p>
+          <mat-divider></mat-divider>
+          <div
+          *cdkVirtualFor="let build of selectedBuilds; "
+          class="build"
+          [class.flaky]="!build.failcount && build.flaky"
+          [class.failing]="build.failcount"
+          [class.success]="!build.failcount && !build.flaky">
```
Comment (chingor13) [inline packages/frontend/src/app/heat-map/heat-map.component.html:31]: In general, I'm not a fan of having too much logic in templates.
